src/components/shop/confirm.js

```diff
@@ -58,21 +69,36 @@ const addConfirmButtons = (scene, container, innerBounds, title, action, item) =
     container.add(confirmButtons);
     ((action === "buy" && !canBuyItem(scene, item)) || (action === "equip" && !isEquippable(item))) &&
         disableActionButton(confirmButtons[0]);
-    resizeConfirmButtons(scene, confirmButtons, innerBounds);
+    sizeConfirmButtons(scene, confirmButtons, innerBounds);
+    return confirmButtons;
 };
 
 export const createConfirm = (scene, title, item) => {
     const action = getAction(scene, title, item);
     scene.title.setTitleText(fp.startCase(action));
+    const confirmPane = {
+        setVisible: fp.noop,
+        action,
+        item,
+        title,
+    };
+    populatePane(scene, confirmPane);
+    confirmPane.resize = resizeConfirm(scene, confirmPane);
+    return confirmPane;
+};
+
+const populatePane = (scene, confirmPane) => {
+    const { action, item, title } = confirmPane;
     const bounds = getSafeArea(scene.layout);
     const container = scene.add.container();
     const innerBounds = getOffsetBounds(bounds, getInnerRectBounds(scene));
     const yOffset = bounds.height / 2 + bounds.y;
     container.setY(yOffset);
     createElems(scene, container, getPromptText({ scene, action, item }), item, innerBounds, bounds);
     action === "buy" && itemIsInStock(scene, item) && createBuyElems(scene, container, item, innerBounds, bounds);
-    addConfirmButtons(scene, container, innerBounds, title, action, item);
-    return container;
+    confirmPane.buttons = addConfirmButtons(scene, container, innerBounds, title, action, item);
+    confirmPane.container = container;
+    confirmPane.destroy = destroyContainer(container);
```

Comment:
try to avoid overriding functions on the pane, this can be hard to keep track of.
if I'm in another file and I see `pane.destroy()`, since its not an object we've defined, it can be confusing to read as it's usually assumed this function came from the library.

a pattern we usually use instead is to return an object containing all the methods you need - eg. like layout.js, collections.js
if you use this pattern instead, when you goto layout.js, etc, you can see all the functions on this object more clearly - instead of any file being able to manipulate injected objects

Reply:
I misread this, it has been changed to be like this, which is better, but I was confused by the confirmPane not being a rexUI pane anymore :D maybe we can attach some of these methods like resize, destroy, etc to the confirmPane initially, or create the object at the end, when we return everything (like layout/collections does, etc)

Reply:
91d6983 hopefully makes things a bit clearer. Wasn't possible to do a completely 'flat' composition at the end of the fn, since the object needs to exist before its `resize` can be created. But now it's a bit more explicit that we're composing a thing, not mutating a thing.

Reply:
bd38887 wrapping it up by updating tests. Nice changes on that pairing call, TY =)